### PR TITLE
 feat(html): Add number of muted findings in HTML report #4703

### LINKED ISSUE #4703: Add number of muted findings in HTML report
### New feature motivation

When you work with the [muted list](https://docs.prowler.com/projects/prowler-open-source/en/latest/tutorials/mutelist/) feature it will be really informative to know how many muted findings you have. Because now even if you mute specific checks they will be marked as failed or passed in the "Assessment Overview" dashboard.
image
image

### Solution Proposed

I would like to see more information in the "Assessment Overview" block. For example:
Total findings: 1859
Passed: 697
Passed (muted): 2
Failed: 1162
Failed (muted): 5
Total Resources: 616

### Describe alternatives you've considered

For now, using bash scripts I parse all muted findings and then output the number of all findings, passed, muted, etc.. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by abant07**
### Context

Currently, the Prowler HTML output displays the total passes and total fails under the Assessment Overview Table. However, it is not clear to a user how many of the total tests and total fails were muted. This PR fixes that issue.

Fix #4703 

### Description

The changes I made were in the `outputs.py` file and the `html.py` file. Specifically, I changed the `extract_findings_statistics` method to look for muted findings, and then created a counter. So now, when the user looks on the prowler dashboard, not only will they see the total number of passed/failed checks, but they will see the number of muted passed and muted failed checks. 

In terms of what I did actually in the code was I just added two new keys to the dictionary that was being returned, and then in the html.py file, I added two new rows in the HTML table, that referenced the muted passed and muted failed keys from the dictionary.

Then, I tested the outputs.py file by adding two new tests `test_extract_findings_statistics_all_passes_are_not_muted` and `test_extract_findings_statistics_all_fail_are_muted`

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/outputs/html/html.py [hunk 1]
```diff
@@ -173,9 +173,15 @@ def write_header(
...
                         <b>Passed:</b> {str(stats.get("total_pass", 0))}
                     </li>
+                    <li class="list-group-item">
+                        <b>Passed (Muted):</b> {str(stats.get("total_muted_pass", 0))}
+                    </li>
                     <li class="list-group-item">
                         <b>Failed:</b> {str(stats.get("total_fail", 0))}
                     </li>
+                    <li class="list-group-item">
+                        <b>Failed (Muted):</b> {str(stats.get("total_muted_pass", 0))}
```
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py]:
```suggestion
                        <b>Failed (Muted):</b> {str(stats.get("total_muted_fail", 0))}
```

File: prowler/lib/outputs/outputs.py [hunk 2]
```diff
@@ -91,24 +91,35 @@ def extract_findings_statistics(findings: list) -> dict:
...
+
         if finding.status == "PASS":
+            if finding.muted is True:
+                muted_pass += 1
             total_pass += 1
             findings_count += 1
+
         if finding.status == "FAIL":
+            if finding.muted is True:
+                muted_fail += 1
```
Comment (jfagoagas) [inline prowler/lib/outputs/outputs.py]: Why not to keep the current logic?
```
if finding.muted is True:
    muted_fail += 1
else:
    if all_fails_are_muted:
        all_fails_are_muted = False
```

File: tests/lib/outputs/html/html_test.py [hunk 3]
```diff
@@ -15,7 +15,9 @@
 
 html_stats = {
     "total_pass": 0,
+    "total_muted_pass": 0,
     "total_fail": 1,
+    "total_muted_fail": 0,
```
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]: The current HTML had a bug, please add a value different than the default one to catch it.
```suggestion
    "total_muted_fail": 10,
```

File: tests/lib/outputs/html/html_test.py [hunk 4]
```diff
@@ -15,7 +15,9 @@
 
 html_stats = {
     "total_pass": 0,
+    "total_muted_pass": 0,
```
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]:
```suggestion
    "total_muted_pass": 20,
```

File: tests/lib/outputs/html/html_test.py [hunk 5]
```diff
@@ -15,7 +15,9 @@
 
 html_stats = {
     "total_pass": 0,
```
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]:
```suggestion
    "total_pass": 5,
```

File: tests/lib/outputs/html/html_test.py
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py:1]: Please adjust the rest of the test with the new values.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
That's great @abant07 👏 

Thanks for this PR to add that feature, please review my comments, they are just quick fixes and some improvements in the tests.

File: prowler/lib/outputs/html/html.py (hunk 1 above)
Comment (abant07) [reply prowler/lib/outputs/html/html.py]: done

File: prowler/lib/outputs/outputs.py (hunk 2 above)
Comment (abant07) [reply prowler/lib/outputs/outputs.py]: done

File: tests/lib/outputs/html/html_test.py (hunk 3 above)
Comment (abant07) [reply tests/lib/outputs/html/html_test.py]: done

File: tests/lib/outputs/html/html_test.py (hunk 4 above)
Comment (abant07) [reply tests/lib/outputs/html/html_test.py]: done

File: tests/lib/outputs/html/html_test.py (hunk 5 above)
Comment (abant07) [reply tests/lib/outputs/html/html_test.py]: done

**Review by jfagoagas — APPROVED** [review]
That's great! Thanks for addressing all the comments.

**Comment by pedrooot** [comment]
Thanks for this! @abant07 🔝

**Pull request merged**